ci-operator/step-registry/ibmcloud/provision/dedicatedhost/static/ibmcloud-provision-dedicatedhost-static-commands.sh

```diff
@@ -0,0 +1,156 @@
+        sleep 10
+        COUNTER=`expr $COUNTER + 20`
+        status=$(${IBMCLOUD_CLI} is dh $dh --output JSON | jq -r ."lifecycle_state")
+        if [[ "${status}" != "pending" ]]; then
+            break
+        fi
+    done
+    echo ${status}
+}
+
+#####################################
+##############Initialize#############
+#####################################
+
+dhProfileMaster="bx2d-host-152x608"
+dhProfileWorker="bx2-host-152x608"
+masterProfile="bx2d-8x32"
+workerProfile="bx2-4x16"
+
+echo "the dh profile of the master nodes: ${dhProfileMaster}; the dh profile of the worker nodes ${dhProfileWorker}"
+
+echo "$(date -u --rfc-3339=seconds) - Creating the Dedicated Host..."
+CLUSTER_NAME="${NAMESPACE}-${UNIQUE_HASH}"
+
+dhName_master="${CLUSTER_NAME}-master-dh"
+dhName_worker="${CLUSTER_NAME}-worker-dh"
+
+dhgRGFile=${SHARED_DIR}/ibmcloud_resource_group
+dhMaster_file=${SHARED_DIR}/dedicated_host_master
+dhWorker_file=${SHARED_DIR}/dedicated_host_worker
+
+#####################################
+#######create the dedicated host#####
+#####################################
+ibmcloud_login
+
+#create the resouce group of the dedicated host group and saved in "${SHARED_DIR}/ibmcloud_resource_group"
+rgDHG1="${CLUSTER_NAME}-rg"
+
+#OCPBUGS-5906 [IPI-IBMCloud] the worker nodes are “Provisioning” when using the dedicated host in the Default group, hide "Default"
+#rg_dhgs=("Default" "${rgDHGMaster}")
+rg_dhgs=("${rgDHG1}")
+selected_idx=$((RANDOM % ${#rg_dhgs[@]}))
+rg_dhg=${rg_dhgs[${selected_idx}]}
+
+if [[ ${rg_dhg} == "Default" ]]; then
+    run_command "${IBMCLOUD_CLI} target --unset-resource-group"
+else    
+    echo "create resource group ... ${rg_dhg}"
+    "${IBMCLOUD_CLI}" resource group-create ${rg_dhg} || exit 1
+    run_command "${IBMCLOUD_CLI} target -g $rg_dhg"
+    echo "${rg_dhg}" >  ${dhgRGFile}
+fi
+
+zone="${region}-${DEDICATEDHOST_ZONE}"
+echo "use the zone defined with DEDICATEDHOST_ZONE: ${zone}"
+
+createDH ${rg_dhg} ${dhProfileMaster} ${zone} ${dhName_master} ${dhMaster_file}
+createDH ${rg_dhg} ${dhProfileWorker} ${zone} ${dhName_worker} ${dhWorker_file}
+
+if [ ! -f ${dhWorker_file} ] || [ ! -f ${dhMaster_file} ]; then
+    echo "Fail to create the dedicated hosts !!"
+    run_command "${IBMCLOUD_CLI} is dhs --resource-group-name ${rg_dhg}"
+    exit 1
+fi
+
+#####################################
+##Create dedicated host yaml file ###
+#####################################
+cat > "${SHARED_DIR}/dedicated_host.yaml" << EOF
+controlPlane:
+  platform:
+    ibmcloud:
+      type: ${masterProfile}
+      zones: ["${zone}"]
+      dedicatedHosts:
+      - name: ${dhName_master}
+compute:
+- platform: 
+    ibmcloud:
+      type: ${workerProfile}
+      zones: ["${zone}"]
+      dedicatedHosts:
+      - name: ${dhName_worker}
+EOF
+
+#OCPBUGS-5906 [IPI-IBMCloud] the worker nodes are “Provisioning” when using the dedicated host in the Default group, or not in the cluster group
+if [[ ${rg_dhg} != "Default" ]]; then
```

Comment:
Even when `rg_dhg` == "Default", will it make any difference to always set it?

Reply:
when the dedicated host group is in Default, need not create the resource group. 
since OCPBUGS-5906, workers fail to deploy when dhg is not in the same group of the cluster.

Reply:
You did not get my points, I was asking why not always set `platform.ibmcloud.resourceGroupName` ?

Reply:
create the cluster in Default Group?  this will make it hard to control the resource

Reply:
Go through your current codes, I did not see any difference with removing the condition.

Reply:
Since bug OCPBUGS-5906, the cluster and the dhg must be in the same group.

Reply:
In your above code, `rg_dhg` stands for the group of the dhg, right? From the descritpion of OCPBUGS-5906, 
```
the worker nodes is “Provisioning” when use the dedicated host in Default group , install the cluster failed. 
```
my understanding is when the dhg is `Default` (cluster rg is defaulted to `Default` too), the bug shows up. Or does your bug mean dhg is `some-new-rg`, cluster rg is defaulted to `Default`, the bug shows up ?

Reply:
yes, rg_dhg is the group of the dhg
I updated the bug title to "[IPI-IBMCloud] fail to retrieve the dedicated host which is not in the cluster group when provisioning the worker nodes"
actually, worker nodes fail to be deployed, when dhg is not in the same resource group of the cluster. 
I have not tested creating the cluster in the "Default" group, shall I do such a test? 
btw, the Default rg is resource group name is "Default"